iceoryx_posh/include/iceoryx_posh/internal/popo/publisher_impl.inl

```diff
@@ -52,7 +53,7 @@ inline expected<AllocationError> PublisherImpl<T, H, BasePublisherType>::publish
                   "callable provided to Publisher<T>::publishResultOf must have signature void(T*, ArgsTypes...)");
 
     return loanSample().and_then([&](auto& sample) {
-        c(sample.get(), std::forward<ArgTypes>(args)...);
+        c(new (sample.get()) T, std::forward<ArgTypes>(args)...);
```

Comment:
This is tricky. If `T` does not have a default ctor this breaks. On the other hand, the code is already broken. Ideally we would have something like Rusts `MaybeUninit<T>` and pass that to the callable.

I admire the problem but don't have a solution.

@elfenpiff what's your opinion on this? Live with the restriction of requiring a default ctor for now and get rid of the undefined behavior?

Reply:
Yeah, I noticed that too. If `T` doesnt have default ctor, then the signature of the callable doesnt make much sense. We could pass in the sample address as `void*`, with the expectation that the callable will placement-new-construct `T` there, but that doesnt play nice with the modern C++ style of iceoryx. Alternatively, we could SFINAE-ize `publishResultOf` so that it is only available when `T` does have default ctor. Or, finally, the callable could just return `T`, which we would again placement-new-copy-construct then, and if the copy-elision mechanisms kick in, that should be efficient. But that would be a breaking public API change...

Reply:
Or rewrite it in Rust! 🚀 😆

Reply:
I think your current solution is okay. Didn't see that @elfenpiff already approved. Since he seems to also be okay with this LGTM